### PR TITLE
move contracts and dependencies to `packages/contracts`

### PR DESCRIPTION
**Pull request by semaraugusto**
Copying typechain files after hardhat build makes it hard to make sure contracts and typechain files being used on the interfaces match. By making this change, there's no need for a typescript package that holds the typescript interfaces anymore and the files will be generated directly by hardhat

### CODE REVIEW

**Comment by semaraugusto** [comment]
TODO: need to add a Semaphore interface as an import for the IdentityVAnchor contract/typescript interface/

File: packages/contracts/test/identityVAnchor/identity_vanchor.test.ts [hunk 1]
```diff
@@ -1020,23 +1017,23 @@ describe('IdentityVAnchor for 2 max edges', () => {
     it('should reject tampering with extData relayer', async () => {
       const invalidExtData = aliceExtData;
       invalidExtData.relayer = toFixedHex('0x0000000000000000000000007a1f9131357404ef86d7c38dbffed2da70321337', 20),
-      await expect(
-        idAnchor.contract.transact({ ...publicInputs }, invalidExtData, { gasLimit: '0x5B8D80' })
-      ).to.be.revertedWith('Incorrect external data hash')
+        await expect(
```
Comment (drewstone) [inline packages/contracts/test/identityVAnchor/identity_vanchor.test.ts:1020]: These spaces seem improperly added and also the format of these tests seems odd?

Why is there no semi-colon and instead a comma straight into a revert check? Maybe some aggregate formatting updates need to be fixed?

File: packages/contracts/test/identityVAnchor/identity_vanchor.test.ts [hunk 2]
```diff
@@ -1020,23 +1017,23 @@ describe('IdentityVAnchor for 2 max edges', () => {
...
+        ).to.be.revertedWith('Incorrect external data hash')
     })
     it('should reject tampering with extData extAmount', async () => {
       const invalidExtData = aliceExtData;
-      invalidExtData.extAmount = toFixedHex(aliceDepositAmount*100, 20),
-      await expect(
-        idAnchor.contract.transact({ ...publicInputs }, invalidExtData, { gasLimit: '0x5B8D80' })
-      ).to.be.revertedWith('Incorrect external data hash')
+      invalidExtData.extAmount = toFixedHex(aliceDepositAmount * 100, 20),
+        await expect(
```
Comment (drewstone) [inline packages/contracts/test/identityVAnchor/identity_vanchor.test.ts:1027]: spacing should be fixed and test formatted properly.

File: packages/contracts/test/vbridge/signatureVBridge.test.ts [hunk 3]
```diff
@@ -755,19 +755,19 @@ describe('2-sided multichain tests for signature vbridge', () => {
...
+        await vBridge.transactWrap(existingToken2.contract.address, [], [ganacheDepositUtxo3], 0, 0, '0', '0', ganacheWallet2);
 
         const webbTokenAddress2 = vBridge.getWebbTokenAddress(chainID2);
         const webbToken2 = await MintableToken.tokenFromAddress(webbTokenAddress2!, ganacheWallet2);
         assert.strictEqual((await webbToken2.getBalance(vAnchor2Address)).toString(), BigNumber.from(7e7).toString());
 
         const balSigners2UnwrappedBefore = await existingToken1.contract.balanceOf(await signers[2].getAddress());
 
-        //Withdraw UTXO 
+        //Withdraw UTXO
```
Comment (drewstone) [inline packages/contracts/test/vbridge/signatureVBridge.test.ts:770]:
```suggestion
        // Withdraw UTXO
```

**Review by drewstone — CHANGES_REQUESTED** [review]

File: .github/workflows/check.yml [hunk 4]
```diff
@@ -27,7 +27,8 @@ jobs:
           node-version: ${{ matrix.node-version }}
       - name: ts-check
         run: |
-          yarn install --frozen-lockfile
+          yarn install
```
Comment (nepoche) [inline .github/workflows/check.yml:30]: Can we keep the frozen lockfile setting?

```suggestion
          yarn install --frozen-lockfile
```

Comment (semaraugusto) [reply .github/workflows/check.yml]: added it back

**Pull request merged**